stackit/internal/core/core.go

```diff
@@ -125,3 +132,21 @@ func LogAndAddErrorBeta(ctx context.Context, diags *diag.Diagnostics, name strin
 	tflog.Error(ctx, fmt.Sprintf("%s | %s", errTitle, errContent))
 	diags.AddError(errTitle, errContent)
 }
+
+func InitProviderContext(ctx context.Context) context.Context {
+	// Capture http response to get trace-id
+	var httpResp *http.Response
+	ctx = runtime.WithCaptureHTTPResponse(ctx, &httpResp)
```

Comment:
```suggestion
	return runtime.WithCaptureHTTPResponse(ctx, &httpResp)
```

No need for an assignment to a variable